test/jdk/javax/swing/JMenuItem/MenuItemTest/MenuItemTestHelper.java

```diff
@@ -168,4 +143,23 @@ private static JMenu createSomeIconsMenu(Icon myIcon, Icon myIcon2) {
 
         return someIcons;
     }
+
+    private record ColoredIcon(Color color, int width,
+                               int height) implements Icon {
+        @Override
+            public void paintIcon(Component c, Graphics g, int x, int y) {
+                Color oldColor = g.getColor();
+                g.setColor(color);
+                g.fillRect(x, y, width, height);
+                g.setColor(oldColor);
+            }
+            @Override
+            public int getIconWidth() {
+                return width;
+            }
+            @Override
+            public int getIconHeight() {
```

Comment:
I'm still for having blank lines between the methods: it makes scanning the code easier as you clearly see where a method ends and another starts.

There's no penalty for additional lines, but they improve readability, in my opinion. In an IDE, you can always collapse unimportant sections, like this helper class, and get it out of your view.